auth.php

```diff
@@ -639,10 +639,23 @@ public function logoutpage_hook() {
     /**
      * Log out user and redirect.
      */
+    public function validate_url() {
+        return filter_var($this->config->redirecturl, FILTER_VALIDATE_URL);
+    }
+
+    public function define_redirect() {
+        global $CFG;
+
+        if (!$this->validate_url()) {
+            return $CFG->wwwroot;
+        }
+        return $this->config->redirecturl;
+    }
+    
     public function user_logout_userkey() {
         global $CFG, $USER;
 
-        $redirect = required_param('return', PARAM_URL);
```

Comment:
Would it be easier to change this line to  `$redirect = required_param('return', PARAM_LOCALURL); ` ?

Reply:
Well observed @dmitriim! Because in this suggestion I thought about the option of preserving the redirection to an external address, according to the url registered in _"Logout redirection URL"_. In this case, I have a question: if we define it as `PARAM_LOCALURL` the redirection will only be possible to addresses related to the Moodle installation, right?

Reply:
>  if we define it as PARAM_LOCALURL the redirection will only be possible to addresses related to the Moodle installation, right?

Yes, that's correct. 

In your patch you're using $this->config->redirecturl which potentially confused you. This config is set in Moodle and used to always redirect a user after logging out from Moodle. E.g. redirecting to IDP logout page so a user is logged out from all apps at once. 

However, in context of this issue $redirect is coning from outside of Moodle. E.g. when an external app requires users to logout from Moodle and then come back to the app for some other actions. 

Hope that makes more sense now.

Reply:
Got it, @dmitriim! Thank you for the clarifications. I will make a new commit with your suggestion.